File: Swift/Longest_Substring_Without_Repeating_Characters.swift
/*Given a string s, find the length of the longest substring without repeating characters.*/


class Solution {
    func lengthOfLongestSubstring(_ s: String) -> Int {
        guard !s.isEmpty else {return 0}
        var j = 0
        var ans = 0
        var h = 0
        var mass:[Character] = []
        for i in s {
            while(mass.contains(i)) {
                mass.remove(at:0);
                j+=1
            }
            mass.append(i)
            ans = max(ans, h - j + 1)
            h+=1
        }
        return ans;
    }
}

File: C++/Longest_Substring_Without_Repeating_Characters.cpp
/*Given a string s, find the length of the longest substring without repeating characters.*/

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if (s.empty()) return 0;
        int i = 0, j = 0, max = 0;
        vector<char> set1;
        for (;i < s.length(); i++) {
            while (find(set1.begin(), set1.end(), s[i]) != set1.end()) {
                set1.erase(set1.begin());
                j++;                                           
            }
            set1.push_back(s[i]);
            if (i - j + 1 > max) {
                max = i - j + 1;
            }
        }
        return max;   
    }
};